python-checks/src/main/java/org/sonar/python/checks/hotspots/HardCodedCredentialsEntropyCheck.java

```diff
@@ -0,0 +1,214 @@
+      .forEach(parameter -> {
+        var value = ((StringLiteral) parameter.defaultValue()).trimmedQuotesValue();
+        patternMatch(parameter.name(), parameter.defaultValue(), value, subscriptionContext);
+      });
+  }
+
+  private void checkDictionaryLiteral(SubscriptionContext subscriptionContext) {
+    var dictionaryLiteral = (DictionaryLiteral) subscriptionContext.syntaxNode();
+    dictionaryLiteral.elements().stream().filter(e -> e.is(Tree.Kind.KEY_VALUE_PAIR))
+      .map(KeyValuePair.class::cast)
+      .filter(keyValuePair -> keyValuePair.value().is(Tree.Kind.STRING_LITERAL))
+      .forEach(keyValuePair -> {
+        var value = ((StringLiteral) keyValuePair.value()).trimmedQuotesValue();
+        var key = keyValuePair.key();
+        if (!key.is(Tree.Kind.STRING_LITERAL)) {
+          return;
+        }
+        patternMatch(((StringLiteral) key).trimmedQuotesValue(), keyValuePair.value(), value, subscriptionContext);
+      });
+  }
+
+  private void checkRegularArgument(SubscriptionContext subscriptionContext) {
+    var regularArgument = (RegularArgument) subscriptionContext.syntaxNode();
+    var keywordArgument = regularArgument.keywordArgument();
+    if (keywordArgument == null) {
+      return;
+    }
+    var expression = regularArgument.expression();
+    if (!expression.is(Tree.Kind.STRING_LITERAL)) {
+      return;
+    }
+    var value = ((StringLiteral) expression).trimmedQuotesValue();
+    patternMatch(keywordArgument, regularArgument, value, subscriptionContext);
+  }
+
+  private void checkAnnotatedAssignment(SubscriptionContext subscriptionContext) {
+    var annotatedAssignment = (AnnotatedAssignment) subscriptionContext.syntaxNode();
+    var assignedValue = annotatedAssignment.assignedValue();
+    var assignedVariable = annotatedAssignment.variable();
+    if (assignedValue == null || !assignedValue.is(Tree.Kind.STRING_LITERAL) || !assignedVariable.is(Tree.Kind.NAME)) {
+      return;
+    }
+    var value = ((StringLiteral) assignedValue).trimmedQuotesValue();
+    patternMatch(((Name) assignedVariable).name(), assignedValue, value, subscriptionContext);
+  }
+
+  private void checkAssignment(SubscriptionContext subscriptionContext) {
+    var assignment = (AssignmentStatement) subscriptionContext.syntaxNode();
+    var assignedValue = assignment.assignedValue();
+    Map<Expression, Name> expressions = null;
+    if (assignedValue.is(Tree.Kind.TUPLE)) {
+      expressions = Expressions.getExpressionsFromRhs((assignment.assignedValue()))
+        .stream()
+        .collect(HashMap::new, (map, e) -> {
+          var name = Expressions.getAssignedName(e);
+          name.ifPresent(value -> map.put(e, value));
+        }, HashMap::putAll);
+    } else {
+      var name = Expressions.getAssignedName(assignedValue);
+      if (name.isPresent()) {
+        expressions = Map.of(assignedValue, name.get());
+      }
+    }
+
+    if (expressions == null) {
```

Comment:
if you will just initialize the `expressions` not as null but as a map and in both condition branches add elements instead of assign it to a new map then you don't need this condition, so entire method could looks like:
```java
private void checkAssignment(SubscriptionContext subscriptionContext) {
    var expressions = new HashMap<Expression, Name>();
    var assignment = (AssignmentStatement) subscriptionContext.syntaxNode();
    var assignedValue = assignment.assignedValue();
    if (assignedValue.is(Tree.Kind.TUPLE)) {
      Expressions.getExpressionsFromRhs(assignedValue)
        .forEach(expression -> {
          var name = Expressions.getAssignedName(expression);
          name.ifPresent(value -> expressions.put(expression, value));
        });
    } else {
      Expressions.getAssignedName(assignedValue)
        .ifPresent(name -> expressions.put(assignedValue, name));
    }

    expressions.entrySet().stream()
      .filter(entry -> entry.getKey().is(Tree.Kind.STRING_LITERAL))
      .forEach(entry -> {
        var expression = (StringLiteral) entry.getKey();
        var name = entry.getValue();
        var value = expression.trimmedQuotesValue();
        patternMatch(name, expression, value, subscriptionContext);
      });
  }
```